core/server/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -449,6 +454,7 @@ private FileInfo getFileInfoInternal(Inode inode) throws FileDoesNotExistExcepti
       throws AccessControlException, FileDoesNotExistException, InvalidPathException {
     MasterContext.getMasterSource().incGetFileInfoOps(1);
     synchronized (mInodeTree) {
+      checkPermission(FileSystemAction.READ, path, false);
```

Comment:
If the path is a file, requiring `Read` permission is okay, but for a directory, we should check `Execute` first.
I think we should remove the `checkPermission` here (getFileInfoInternal will check it) and add `Execute` permission check at `inode.isDirectory()`